mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -43,6 +55,379 @@ struct CompactBlockRow {
+            let block = self.get_block_by_height(height).await?;
+            debug!("Got block {:?}", block);
+            let mut compact_txs = Vec::new();
+            // By default, CompactBlocks only contain CompactTxs for transactions that contain Sapling spends or outputs.
+            // Create and push compact_tx during iteration.
+            for (tx_id, hash_tx) in block.tx.iter().enumerate() {
+                let tx_bytes = self.get_transaction_bytes(hash_tx).compat().await?;
+                let tx = ZTransaction::read(tx_bytes.as_slice()).unwrap();
+                let mut spends = Vec::new();
+                let mut outputs = Vec::new();
+                if !tx.shielded_spends.is_empty() || !tx.shielded_outputs.is_empty() {
+                    // Create and push spends with outs for compact_tx during iterations.
+                    for spend in &tx.shielded_spends {
+                        let compact_spend = TonicCompactSpend {
+                            nf: spend.nullifier.to_vec(),
+                        };
+                        spends.push(compact_spend);
+                    }
+                    for out in &tx.shielded_outputs {
+                        let compact_out = TonicCompactOutput {
+                            cmu: out.cmu.to_bytes().to_vec(),
+                            epk: out.ephemeral_key.to_bytes().to_vec(),
+                            // https://zips.z.cash/zip-0307#output-compression
+                            // The first 52 bytes of the ciphertext contain the contents and opening of the note commitment,
+                            // which is all of the data needed to spend the note and to verify that the note is spendable.
+                            ciphertext: out.enc_ciphertext[0..52].to_vec(),
+                        };
+                        outputs.push(compact_out);
+                    }
+                    // Shadowing mut variables as immutable. No longer need to update them.
+                    drop_mutability!(spends);
+                    drop_mutability!(outputs);
+                    let mut hash_tx_vec = hash_tx.0.to_vec();
+                    hash_tx_vec.reverse();
+
+                    let compact_tx = TonicCompactTx {
+                        index: tx_id as u64,
+                        hash: hash_tx_vec,
+                        fee: 0,
+                        spends,
+                        outputs,
+                    };
+                    compact_txs.push(compact_tx);
+                }
+            }
+            let mut hash = block.hash.0.to_vec();
+            hash.reverse();
+            // Set 0 in vector in the case of genesis block.
+            let mut prev_hash = block.previousblockhash.unwrap_or_default().0.to_vec();
+            prev_hash.reverse();
+            // Shadowing mut variables as immutable.
+            drop_mutability!(hash);
+            drop_mutability!(prev_hash);
+            drop_mutability!(compact_txs);
```

Comment:
On the one hand, I agree with you, because that is how rust works.
 On the other hand, using that macros we can avoid future bugs or ub (reference to [this comment](https://github.com/KomodoPlatform/atomicDEX-API/pull/1327#pullrequestreview-1025882136)). We can just be sure, that everything will be ok.
In addition `drop_mutability!` is zero cost, so its only about readability issue.
 
We could just have the project rule "if you don't need the mutable variable - drop the mutability".